pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -132,6 +132,13 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		if !isCreated(replicas[i]) {
 			return ssc.podControl.CreateStatefulPod(set, replicas[i])
 		}
+		// If we find a Pod that is currently terminating, we must wait until graceful deletion
+		// completes before we continue to make  progress.
+		if isTerminating(replicas[i]) {
+			glog.V(2).Infof("StatefulSet %s is waiting for Pod %s to Terminate",
```

Comment:
I think the log statements in this function are informational. Informational logs should meet the criteria of being useful to an administrator for diagnostics and mitigation on a running cluster. If a StatefulSet has been deployed, and if it is not making progress due to a Pod's condition, imo, this is useful to surface to an administrator via logs. 
I agree that the exhaust we produce in the controller manager logs make it difficult to discriminate what is useful vs what is not, and we need to look across the controllers at large and apply the existing, or another, convention to apply better filtering to our exhaust.